canonical/snapcraft.io
# Wd 14634 metrics page

## Done
- Added endpoints to fetch active devices metrics, annotation information and territory metrics.
- Migrated Metrics page to React.
- Added unit tests for endpoints and components.
- Added error state to the metrics page.
- Added loading state.

## How to QA
- Go to https://snapcraft-io-4851.demos.haus/lxd/metrics
- Verify the graphs loads
- change "by version" to something else
- Verify the weekly active devices graph changes
- change the "Past 30 days" to "Past 7 days"
- Verify the weekly active devices graph changes
- Refresh the page
- Verify the filters are applied correctly
- change the "Past 7 days" to "Past 1 year"
- Verify the error view is shown
- Go to https://snapcraft-io-4851.demos.haus/misk-test/metrics
- Verify no data view is shown
- Verify the filters are disabled.

## Testing

## Issue / Card
Fixes [#14634](https://warthogs.atlassian.net/browse/WD-14634)

## Screenshots
![Screenshot from 2024-09-20 10-32-52](https://github.com/user-attachments/assets/3f0fd8be-b90d-4c5f-8aca-616d28eb8131)
![Screenshot from 2024-09-20 10-31-18](https://github.com/user-attachments/assets/5ca3ce77-baa8-4a3c-88dc-b45d5b43c5e6) 
![Screenshot from 2024-09-20 10-32-03](https://github.com/user-attachments/assets/e1a1bab6-1de1-46b2-a331-5c88e631dacf)
![Screenshot from 2024-09-20 10-32-19](https://github.com/user-attachments/assets/323fc167-4858-4cb0-9bae-233262c2f7b3)
